services/director-v2/tests/unit/with_swarm/test_modules_dynamic_sidecar_docker_api.py

```diff
@@ -569,3 +600,30 @@ async def test_is_dynamic_service_running(
         await docker_api.is_dynamic_service_running(node_uuid, dynamic_sidecar_settings)
         is True
     )
+
+
+async def test_get_project_networks_containers(
+    async_docker_client: aiodocker.docker.Docker,
+    project_id_labeled_network: str,
+    project_id: ProjectID,
+    docker_swarm: None,
+) -> None:
+    # make sure API does not change
+    params = {"filters": clean_filters({"label": [f"project_id={project_id}"]})}
+    filtered_networks = (
+        # pylint:disable=protected-access
+        await async_docker_client.networks.docker._query_json("networks", params=params)
+    )
+    assert len(filtered_networks) == 1
+    filtered_network = filtered_networks[0]
+
+    assert project_id_labeled_network == filtered_network["Id"]
+
+
+async def test_get_or_create_networks_ids(
+    test_networks: List[str], project_id: ProjectID
+):
+    network_ids = await docker_api.get_or_create_networks_ids(
+        networks=test_networks + test_networks, project_id=project_id
+    )
+    assert set(test_networks) == set(network_ids.keys())
```

Comment:
here you are just testing the the fct runs right? you do not get them but always create them am I right? and you do not know actually if the networks are really created.

Reply:
A few things are done. If the networks exist, they are fetched and the ids are returned. If the networks are missing they are crated and the ids are fetched and returned.
The last step is fetch the ids and return them. See `services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py`

Reply:
what I meant is that this very test only creates them, but does not test the getting.

Reply:
OK, extended test to include this case as well.